transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -203,6 +207,28 @@ void add(AbstractEpollChannel ch) throws IOException {
     void modify(AbstractEpollChannel ch) throws IOException {
         assert inEventLoop();
         Native.epollCtlMod(epollFd.intValue(), ch.socket.intValue(), ch.flags);
+        ch.activeFlags = ch.flags;
+    }
+
+    void updatePendingFlagsSet(AbstractEpollChannel ch) {
+        pendingFlagChannels.set(ch.socket.intValue(), ch.flags != ch.activeFlags);
+    }
+
+    private void processPendingChannelFlags() {
+        // Call epollCtlMod for any channels that require event interest changes before epollWaiting
+        if (!pendingFlagChannels.isEmpty()) {
+            for (int fd = 0; (fd = pendingFlagChannels.nextSetBit(fd)) >= 0; pendingFlagChannels.clear(fd)) {
+                AbstractEpollChannel ch = channels.get(fd);
+                if (ch != null) {
+                    try {
+                        ch.modifyEvents();
+                    } catch (IOException e) {
+                        ch.pipeline().fireExceptionCaught(e);
+                        ch.close(ch.voidPromise());
```

Comment:
voidPromise() is not allowed for `ch.close(...)`. Just call `ch.close()`.

Reply:
@normanmaurer I was just keeping this the same as the equivalent removed logic in these two places: https://github.com/netty/netty/pull/9397/files#diff-a1f23435bb30b3ac00bf72d14e710928L455 and https://github.com/netty/netty/pull/9397/files#diff-a1f23435bb30b3ac00bf72d14e710928L540. Could you confirm that those are/were incorrect?

Reply:
using `unsafe.close(voidPromise)` is ok... while `channel.close(voidPromise)` is not.

Reply:
Done, but then I just noticed that `ch.unsafe().close(ch.unsafe().voidPromise())` is done elsewhere in the same class: https://github.com/netty/netty/blob/517a93d87d5c65e3c4bedf4a77183583991f32e1/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java#L382

want me to do that instead?

Reply:
The question is should the close traverse the pipeline or not... ?

Reply:
Ah right, then I guess may as well leave as-is if you agree.